p2p/peer.py

```diff
@@ -726,16 +726,16 @@ def _get_random_bootnode(self) -> Generator[Node, None, None]:
             # as discussed in
             # https://github.com/ethereum/py-evm/pull/139#discussion_r152067425
             peer = await self.connect(node)
-            if self.is_full:
+
+            if peer is None:
+                continue
+            elif self.is_full:
```

Comment:
We should simply refrain from attempting to self.connect() when we're full -- it's only for incoming connections that we should disconnect after the handshake.

Reply:
will work on getting this cleaned up today.